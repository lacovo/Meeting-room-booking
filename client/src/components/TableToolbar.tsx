import React, { useState } from 'react';
import {
  Toolbar,
  Typography,
  IconButton,
  Tooltip,
  FormControlLabel,
  Switch,
} from '@material-ui/core';
import { Delete, FilterList, Edit } from '@material-ui/icons';
import { useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { setMatch } from '../redux';
import { IAppState, IMatch } from '../types';
import Filter from './Filter';

export default function EnhancedTableToolbar(props: any) {
  const { selected, deleteData } = props;
  const numSelected = selected.length;
  const [open, setOpen] = useState(false);
  const history = useHistory();
  const dispatch = useDispatch();
  const match: IMatch = useSelector((state: IAppState) => {
    return state.match;
  });

  const handleChangeAll = (eventprops: any) => {
    const thismatch = { ...match, is_all: !eventprops.target.checked };
    dispatch(setMatch(thismatch));
  };
  const handleOpenFilter = () => {
    setOpen(!open);
  };

  const handleCloseFilter = () => {
    setOpen(false);
  };
  const handleDelete = () => {
    deleteData();
  };

  return (
    <Toolbar>
      {numSelected > 0 ? (
        <Typography color="inherit" variant="subtitle1" component="div">
          {numSelected} selected
        </Typography>
      ) : (
        <Typography variant="h6" id="tableTitle" component="div">
          Bookings
        </Typography>
      )}

      {numSelected > 0 ? (
        <Tooltip title="Delete">
          <IconButton onClick={handleDelete}>
            <Delete />
          </IconButton>
        </Tooltip>
      ) : (
        <>
          <Tooltip title="Filter list">
            <IconButton onClick={handleOpenFilter}>
              <FilterList />
            </IconButton>
          </Tooltip>
          <Filter open={open} handleClose={handleCloseFilter} />
        </>
      )}
      {numSelected === 1 && (
        <Tooltip title="Edit">
          <IconButton
            onClick={() => history.push(`/roomdetail/${selected[0]}`)}
          >
            <Edit />
          </IconButton>
        </Tooltip>
      )}
      <FormControlLabel
        className="switch-all"
        control={<Switch defaultChecked onChange={handleChangeAll} />}
        label="Display valid booking"
      />
    </Toolbar>
  );
}
