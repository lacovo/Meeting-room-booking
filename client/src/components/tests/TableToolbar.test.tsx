import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TableToolbar from '../TableToolbar';
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: () => {},
  useSelector: () => {
    return [{   status: 0,
      match: {roomId: "", roomName: "", date:"", host: "", guest: "", orderBy: "default", is_all: false},
      bookings: [],
      rooms: [], }];
  },
}));

describe('<TableToolbar />', () => {
  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<TableToolbar selected={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

