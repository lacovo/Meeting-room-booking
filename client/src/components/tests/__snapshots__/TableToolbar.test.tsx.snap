// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableToolbar /> Should render and match the snapshot 1`] = `
<WithStyles(ForwardRef(Toolbar))>
  <WithStyles(ForwardRef(Typography))
    component="div"
    id="tableTitle"
    variant="h6"
  >
    Bookings
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Tooltip))
    title="Filter list"
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo(ForwardRef(FilterListIcon)) />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Tooltip))>
  <Filter
    handleClose={[Function]}
    open={false}
  />
  <WithStyles(ForwardRef(FormControlLabel))
    className="switch-all"
    control={
      <ForwardRef(WithStyles)
        defaultChecked={true}
        onChange={[Function]}
      />
    }
    label="Display valid booking"
  />
</WithStyles(ForwardRef(Toolbar))>
`;
